src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -1744,6 +1744,13 @@ static void updateAttrInNotifyCer
 
   /* Reached this point, it means that it is a new attribute (APPEND case) */
   ContextAttribute* caP = new ContextAttribute(targetAttr, useDefaultType);
+
+  if (caP->compoundValueP)
+  {
+    // The ContextAttribute constructor steals the compound, but in this case, it must be cloned
+    targetAttr->compoundValueP = caP->compoundValueP->clone();
```

Comment:
Discussed at skype. Seems to be leak-safe from in principle (valgrind job will confirm) but the following comment in 'updateAttrInNotifyCer' should be adjusted:

```
  // Note that updateAttrInNotifyCer() may "ruin" targetAttr, as compoundValueP is moved
  // (not copied) to the structure in the notifyCerP and null-ified in targetAttr. Thus, it has
  // to be called after the location processing logic (as this logic may need the compoundValueP
```